hop/subscribe.py

```diff
@@ -1,64 +1,64 @@
 #!/usr/bin/env python
 
 __author__ = "Bryce Cousins (bfc5288@psu.edu)"
-__description__ = "tools to receive and parse GCN circulars"
+__description__ = "tools to receive and parse messages"
 
 
 import argparse
 import json
 
 from . import cli
 from .io import Stream
-from .models import GCNCircular, VOEvent
+from .models import GCNCircular, VOEvent, message_blob
 
 
-def classify_msg(msg):
-    """Check and classify the format of a message obtained from an ADC
-    stream, use it to instantiate a data model corresponding to that format.
+def classify_message(message):
+    """Check the format of a message obtained from an ADC stream and
+    use it to instantiate a data model corresponding to that format.
 
     Args:
-      msg:       raw message from an ADC stream
+      message:       wrapped message from an ADC stream
 
     Returns:
-      msg_model: dataclass model object for the raw message
+      message_model: dataclass model object for the message
 
     """
-    # check for msg format using standard-specific flags
-    voevent_flag = "ivorn"
-    gcncir_flag = "GCN CIRCULAR"
-
-    # VOEvent:
-    if voevent_flag in msg:
-        msg_model = VOEvent(**msg)
-        status_str = "## Parsing a VOEvent"
-    # GCN circular:
-    elif gcncir_flag in msg:
-        msg_model = GCNCircular(**msg)
-        status_str = "## Parsing a GCN Circular"
-    else:
-        msg_model = GCNCircular(**msg)
-        status_str = "## Parsing a hop-published message"
 
-    print(status_str)
+    try:
+        fmt = message["format"]
+        content = message["content"]
+    except TypeError:
```

Comment:
I think also want to check for `KeyError`, in case `message` is a dict but it just doesn't have those keys.

Reply:
Good catch, I added this and a test for both errors.